curriculum/challenges/english/07-scientific-computing-with-python/learn-list-comprehension-by-building-a-case-converter-program/657efa642593c5746acc5c81.md

```diff
@@ -18,11 +18,7 @@ You should assign the modified character to a variable named `converted_characte
 ```js
 ({
     test: () => {
-        const transformedCode = code.replace(/\r/g, "");
-        const convert_to_snake_case = __helpers.python.getDef("\n" + transformedCode, "convert_to_snake_case");
-        const { function_body } = convert_to_snake_case;
-
-        assert.match(function_body, / +converted_character\s*=\s*("|')_\1\s*\+\s*char\.lower\s*\(\s*\)/);
+     assert(runPython(`_Node(_code).find_function("convert_to_snake_case").find_for_loops()[0].find_body().find_ifs()[0].find_bodies()[0].find_variable('converted_character').is_equivalent("converted_character = '_' + char.lower()") or _Node(_code).find_function("convert_to_snake_case").find_for_loops()[0].find_body().find_ifs()[0].find_bodies()[0].find_variable('converted_character').is_equivalent("converted_character = f'_{char.lower()}'")`)); 
```

Comment:
```suggestion
assert(runPython(`
  converted_character_variable = _Node(_code).find_function("convert_to_snake_case").find_for_loops()[0].find_ifs()[0].find_variable('converted_character')
  
  converted_character_variable.is_equivalent("converted_character = '_' + char.lower()") or converted_character_variable.is_equivalent("converted_character = f'_{char.lower()}'")
`));
```
Trying to make it a bit simpler.


I've also thrown it into python formatter (https://black.vercel.app/) 👀 :
```suggestion
    assert(runPython(`
      converted_character_variable = (
          _Node(_code)
          .find_function("convert_to_snake_case")
          .find_for_loops()[0]
          .find_ifs()[0]
          .find_variable("converted_character")
      )
      
      converted_character_variable.is_equivalent(
          "converted_character = '_' + char.lower()"
      ) or converted_character_variable.is_equivalent(
          "converted_character = f'_{char.lower()}'"
      )
    `)); 
```